packages/main/src/Select.ts

```diff
@@ -418,12 +426,12 @@ class Select extends UI5Element {
 			};
 		});
 
-		if (lastSelectedOptionIndex > -1 && !syncOpts[lastSelectedOptionIndex].disabled) {
+		if (lastSelectedOptionIndex > -1 && !(syncOpts[lastSelectedOptionIndex] as Option).disabled) {
```

Comment:
I would prefer to create new type for syncOpts (something like InputSuggestionText) because casting it to option is not correct and that check with disabled property seems to be redundant since it will never exist

Reply:
syncOpts is not array of Option (as Nayden said), although they share similar props, the Option extends Ui5 Element, HTMLElement and so on. So this is wrong in that regards. We need to declare a new type.

I saw it's used for the entries in the List in the Select Popover, so it seems that these fields on the syncedOptions are the things required for an option to be properly rendered as a list item. So it could be also interface IOption, and the Option component itself can implement it.